### PR TITLE
don't corrupt version negotiation packets

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc [hunk 1]
```diff
@@ -30,39 +32,22 @@ void CorruptRateErrorModel::DoReset(void) { }
...
-    uint8_t *buffer= new uint8_t[p_len];
-    p->CopyData(buffer, p_len);
-    PppHeader ppp_hdr = PppHeader();
-    uint32_t ppp_hdr_len = p->RemoveHeader(ppp_hdr);
-    Ipv4Header ip_hdr = Ipv4Header();
-    uint32_t ip_hdr_len = p->RemoveHeader(ip_hdr);
-    UdpHeader udp_hdr = UdpHeader();
-    uint32_t udp_hdr_len = p->RemoveHeader(udp_hdr);
-    const uint32_t hdr_len = ppp_hdr_len + ip_hdr_len + udp_hdr_len;
+    QuicPacket qp = QuicPacket(p);
```
Comment (janaiyengar) [inline sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc]: Nice refactor!

File: sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc [hunk 2]
```diff
@@ -30,39 +32,22 @@ void CorruptRateErrorModel::DoReset(void) { }
...
+    // Don't corrupt Version Negotiation packets.
+    // Version Negotiation packets are expected to be sent when the version field of the Initial was corrupted.
+    // Client are supposed to ignore Version Negotiation packets that contain the version that they offered.
+    if(qp.IsVersionNegotiationPacket()) {
+        qp.ReassemblePacket();
+        return false;
```
Comment (janaiyengar) [inline sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc]: Why is this necessary? If the VN packet is corrupted, the client is likely to drop the packet, right?

Comment (marten-seemann) [reply sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc]: This requires 2 things to happen:
1. The version number field on the client's Initial is corrupted, eliciting a VN packet from the server.
2. The VN packet will now contain the (uncorrupted) version that the client selected. This tells the client that it can ignore the VN packet. But if this version number in the VN packet now is *also* corrupted, the client would accept the VN packet and the connection attempt will fail.

Comment (marten-seemann) [reply sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc]: @janaiyengar See https://github.com/quicwg/base-drafts/pull/3533.

Comment (huitema) [reply sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc]: I think that's a good idea. Corrupted VN generate random failures, which distracts from the purpose of the test.

Comment (marten-seemann) [reply sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc]: I'll take this as an approval of this PR 🙃
